### ISSUE TITLE
[New build - DEFAULT] 20w20a: Starting Zone Tutorial

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2011]
Tutorial starting zones have been added to the game.
They are made using the game logic.

Changes:
 - Preview for the starter block
 - Tutorial zone added to the starting point
 - Help message popup windows mode
 - Help messages can use sprite

Bug Fixes:
#1991 Wrong Canopy hitbox.
#1993 Stronger player gravity field.
Collision for Bast and Trap door.
Secondary paint for Blast and Trap door.
Wrong 3d zone for the volumetric sensor

**Comment by tsunamayo** [comment]
Build is live - btw I am very much looking forward from feedback on the starting zone, so hopefully newcomers have a smooth first contact with the game!
=> any broken things?
=> any more stuff I should explain
=> any misspell ect?
Cheers

**Comment by Majikmonster** [comment]
Immediately, I see that the new window pop-ups are broken. All the wonderful text and sprites in the message brick don't show up in the pop-up, and are replaced with the words "New Text".

**Comment by tsunamayo** [comment]
all of them? It must be a build only bug, because I have spent the whole 2 day using them :p 
I keep you posted...

**Comment by tsunamayo** [comment]
okay fixed, this was a bug with TextMeshPro. I am pushing a new build, I keep you guys posted.

**Comment by tsunamayo** [comment]
okay build is up again, now you can check the tuts ;)

**Comment by Majikmonster** [comment]
Latest build I checked has no text in the message bricks. The input fields for the gates are empty :(

**Comment by Dwarf-LordPangolin** [comment]
So I went and played through the tutorial; here's my feedback on the play element. I'll give separate feedback on spelling/grammar later.

- Building 1 tells you to switch to using anti-grav movement, but the player starts in that mode when the tutorial begins. Guessing the player is meant to start the tutorial in Walk instead.
- The Push Tool minigame is a great idea, but the text panel perhaps should include directions on how to get it the tool from the item stand.
- The door in the 6dof mode tutorial starts open; not sure if bug or intentional.
- The section demonstrating the difference between anti-grav and 6dof is inspired; excellent choice there!
- When the player leaves building 1, they'll probably still be in anti-grav or 6dof, but the way out is an elevator. Not a bug or problem, but I anticipate some players being confused for a moment when the elevator starts moving down and they're still hovering in mid-air. ;)
- The red stairs up to the build area are too steep to walk up, and the player cannot jump high enough to get up them.
- The tooltip tells you to rotate the brick with the R key, but Q and E are the rotate keys.
- When new blocks are picked up, they'll appear in whatever the currently selected slot on the hotkey bar is. This means that picking up the starter block, hull block, and wedge block simply results in changing what is in that selected slot 3 times. Is it possible to make newly picked up items occupy unused slots on the toolbar instead?

Overall it looks like the tutorial covers all the basics of building, which is great! I noticed there wasn't anything about how landing at stations or spawning NPCs works; do you want to add more tutorial stuff later to cover those?

**Comment by Womble-TC** [comment]
Hey Tsuna, I'm still working on the tutorial building(s) you asked for a few days before this update, if you want, I can rework it to include all of the stuff in the current tutorial. I've been getting some help building, so it should take too long (under a week hopefully). Here is building one near completion: 
![image](https://user-images.githubusercontent.com/29961381/81599876-c79a3f00-93c0-11ea-8170-0b4e64246ec8.png)
![image](https://user-images.githubusercontent.com/29961381/81599919-d08b1080-93c0-11ea-8a82-81c6a6359caa.png)
https://cdn.discordapp.com/attachments/353281519357263874/709475065468813383/unknown.png

I can copy and paste all of the monitors and hud pop-ups, so it shouldn't take long to copy everything over. If you still want to use your buildings however, that's fine. (Sorry I wasn't quick enough with my building!)

**Comment by tsunamayo** [comment]
@Womble-TC yeah it is look pretty cool, send it my way I will see if I can integrate it!
Now the thing is for ship building I realize that it must be something outdoor, with the ship in the middle as player are gonna build on it at each step, so I am not sure if the closed one could work. Maybe in replacement of the last section?

**Comment by Womble-TC** [comment]
I was thinking this would be for the first building, (building 1) and then they can use a hovercraft in this building's garage to get to the next area. I can send it to you in a bit.

**Comment by Womble-TC** [comment]
@tsunamayo here it is 
[pad 1.zip](https://github.com/tsunamayo/Starship-EVO/files/4636837/pad.1.zip)
This is for the movement tutorial section, here are some pics: 
![image](https://user-images.githubusercontent.com/29961381/82090201-0d227900-96ed-11ea-88be-34150d005363.png)
![image](https://user-images.githubusercontent.com/29961381/82090215-13185a00-96ed-11ea-9261-d83f1a7b93d6.png)
https://cdn.discordapp.com/attachments/416013956331601930/710939632082092132/unknown.png
hope you can find space for it somewhere! Me and the others can do the other sections too if you need it.

**Comment by Womble-TC** [comment]
Oh, and you will need to enter using the hatch on the roof.
![image](https://user-images.githubusercontent.com/29961381/82090469-8fab3880-96ed-11ea-8f5d-8fc750cf245e.png)

**Comment by JRL101** [comment]
BUG: 20w20d 
Character sliding on elevator when in gravity. Seem to be getting pushed to the sides, noticed consistently in the **tutorial area 1**

**Comment by JRL101** [comment]
**[Controls] R to rotate?** 
I only seem to be able to rotate blocks with Q/E 
I don't see an option for R in the key bindings either.

Build Zone 2 (beginning)
![image](https://user-images.githubusercontent.com/33255536/82121326-c1400480-97be-11ea-8767-badd94a966f8.png)

**Comment by tsunamayo** [comment]
@Womble-TC Sorry I am just checking your build now and it looks amazing! I will use it for the upcoming wave.

Now yes I would very much like if you guys could work on the two other zone!

Zone 2) I think it would need more spacious and more enclosed. So it would be an octogon like the current one, with each side being a part of the tutorial. The middle part would left to open air so the player can exit the their first ship. The rest of the room could be enclosed. You can make a ramp to access it, as I understood player would use a hovercraft from the ramp in Zone 1.

Zone 3) So that one could be something enclosed too. Maybe like the old tutorial? Or maybe a small tower with a tut for each level? In any case I will likely need to spawn 1 (or 2) more in the future when we have more features to explain. 

This wont be on time for the next wave but if we could get it for the Early Access that would be amazing!

Your style is doing wonder, great detailing but not too much (which is important for performance). End result is super clean!

**Comment by JRL101** [comment]
@tsunamayo Would you mind if i worked on a tutorial area too? just as an interesting project.
